clients/src/main/java/org/apache/kafka/common/MetricName.java

```diff
@@ -75,53 +74,6 @@ public MetricName(String name, String group, String description, Map<String, Str
         this.tags = Utils.notNull(tags);
     }
 
-    /**
-     * @param name          The name of the metric
-     * @param group         logical group name of the metrics to which this metric belongs
-     * @param description   A human-readable description to include in the metric
-     * @param keyValue      additional key/value attributes of the metric (must come in pairs)
-     */
-    public MetricName(String name, String group, String description, String... keyValue) {
```

Comment:
MetricName constructors are part of the public api. So, we probably can't just remove existing ones. If we really want to remove them, we will need to mark them as deprecated first and remove them a few releases later.